core/src/main/java/nl/inl/blacklab/contentstore/ContentStoreFixedBlockWriter.java

```diff
@@ -249,20 +257,34 @@ private void writeBlocks(boolean writeLastBlock) {
         ensureContentsFileOpen();
 
         // Do we have a block to write?
-        while (writeLastBlock && unwrittenContents.length() > 0
-                || unwrittenContents.length() >= WRITE_BLOCK_WHEN_CHARACTERS_AVAILABLE) {
-            int lenBefore = unwrittenContents.length();
+        while (writeLastBlock && getUnwrittenByteCount() > 0 || getUnwrittenByteCount() >= WRITE_BLOCK_WHEN_CHARACTERS_AVAILABLE) {
+            int offsetBefore = unwrittenIndex;
             byte[] encoded = encodeBlock(); // encode a number of characters to produce a 4K block
-            int lenAfter = unwrittenContents.length();
-            int charLen = lenBefore - lenAfter;
+            int offsetAfter = unwrittenIndex;
+            int charLen = offsetAfter - offsetBefore;
             int blockIndex = writeToFreeBlock(encoded);
             blockIndicesWhileStoring.add(blockIndex);
             blockCharOffsetsWhileStoring.add(charsFromEntryWritten);
             charsFromEntryWritten += charLen;
             bytesWritten += encoded.length;
+            
+            if (unwrittenIndex > MAX_UNWRITTEN_INDEX) {
+                this.unwrittenContents.delete(0, unwrittenIndex);
+                unwrittenIndex = 0;
+            }
+        }
+
+        // Free memory
+        if (unwrittenIndex > 0 && getUnwrittenByteCount() == 0) {
```

Comment:
Waarom unwrittenIndex > 0? Wil je hier niet gewoon zeggen "als het hele document geschreven is, free dan het memory"?

Reply:
Ja, ik denk dat die inderdaad niet helemaal klopt. Hij zal nooit per ongelijk data weggooien, maar wat er eigenlijk getest moet worden is of de StringBuilder's interne array groot is (omdat er groot document is gepasseerd dat inmiddels al helemaal is opgeslagen), en zo ja dan even de stringbuilder weggooien om die array te laten garbage collecten.

Reply:
Als het goed is gefixt.